flow/render_status.py

```diff
@@ -0,0 +1,220 @@
+# make separate python class for render_status
+import mistune
+from .scheduling.base import JobStatus
+
+
+class renderer:
```

Comment:
Could you briefly elaborate on what the exact purpose of this class is? It seems to introduce a lot of boilerplate and is probably responsible for pushing this PR into a net positive increase in code lines, whereas the main purpose of this PR is to simplify and reduce code size.

Reply:
I think the main idea is to separate the rendering logic into this class so the the pring_status only deal with gathering the necessary job info for renderer class to render. This way we can have a clear place for rendering code and for future extension if necessary. Also this makes it possible to provide the whole status output class object for user to use and modify without digging into the main project.py code base. @vyasr I think we also discussed this a while a ago, any thoughts?

Reply:
I think this is generally a good idea, however the class structure appears highly intransparent to me at the moment. Would you mind checking if this could potentially be refactored? Maybe a set of functions would be more appropriate?

I could also give it a pass, just let me know, but I'd prefer if you could have critical look at this yourself first.

Reply:
I think the main reason for the increase is that there are a lot more comment lines now because all of the options to `print_status` are duplicated in the docstring of `renderer.render`. The total lines of code are reduced. However, I'm happy to look at this with Pengji to find possible simplifications, we could definitely benefit from simplifying this code further.